### ISSUE TITLE
Subcommand help text lacks required parent command arguments

### ISSUE THREAD

**Issue by symbiont-joseph-kachmar** [pcapriotti/optparse-applicative#359]
I'm currently using `optparse-applicative` to construct a CLI parser where the command tree has top-level commands that take required and optional inputs as well as subcommands that may make use of the inputs given to the parent command.

As far as I can tell, there's no way to have `optparse-applicative` generate help text and/or example usage that reflects the required input coming from the parent command parser.

---

Take the following program as an example, where I'm constructing an options parser for a parent command and a subcommand that allows subcommand's options to be passed inline with the parent command's.

https://gist.github.com/symbiont-joseph-kachmar/84a7e47fba5a7f1cc46c8603bd916f92

If I compile this to some executable `exe` and call `exe --help`, I get the following output:
```
Usage: exe (-a|--arg COMMAND_INPUT) [COMMAND]
  Optparse subcommand example...

Available options:
  -h,--help                Show this help text
  -a,--arg COMMAND_INPUT   Required command input

Available commands:
  sub                      Example subcommand
```
If I call `exe sub --help`, however, the "Available options" stanza makes no mention of the required `COMMAND_INPUT` option:
```
Usage: exe sub (-s|--sub-arg SUBCOMMAND_ARG)
  Example subcommand

Available options:
  -h,--help                Show this help text
  -s,--sub-arg SUBCOMMAND_ARG
                           Required subcommand input
```
Finally, if I call `exe sub`, the error response will correctly tell me which options were missing, but (again) these options won't be shown in the usage section:
```
Missing: (-a|--arg COMMAND_INPUT) (-s|--sub-arg SUBCOMMAND_ARG)

Usage: exe sub (-s|--sub-arg SUBCOMMAND_ARG)
  Example subcommand

Available options:
  -h,--help                Show this help text
  -s,--sub-arg SUBCOMMAND_ARG
                           Required subcommand input
```

---

I imagine that due to the nature of how `Parser`s are constructed, this type of thing would probably be difficult/impossible to support directly right now (I can't think of a way for the subcommand `Parser` to "know" about the required options of its parent).

What might be necessary to add support for this to the library; barring that, are there any patterns or approaches I could take to manually add support for this behavior?

If the relevant `Doc`s generating this text could be pulled out from the parent `Parser` I could pass that to the function constructing the subcommand `Parser` and I think that would be fine.

**Comment by HuwCampbell** [comment]
Yeah... this is a bit tricky.

In inline mode we might be able to do better. I'll have a harder think and a go, but we might be able to use the full inlined parser as the context.

**Comment by HuwCampbell** [comment]
Please see #385.

This is also a duplicate of #294

**Comment by HuwCampbell** [comment]
Fixed in #385
